accepted/0000-install-size-info.md

```diff
@@ -0,0 +1,201 @@
+```
+
+#### `npm install --json`
+
+The JSON format will add two things: `size` and `unpackedSize` field to each
+individual `action` record, and a `sizeDelta` field that totals up all the
+`unpackedSize` changes, in the toplevel:
+
+```
+➜ npm i ssri --dry-run --json
+{
+  "added": [
+    {
+      "action": "add",
+      "name": "ssri",
+      "version": "6.0.0",
+      "path": ".../node_modules/ssri",
+      "size": 12116,
+      "unpackedSize": 40772
+    }
+  ],
+  "removed": [],
+  "updated": [],
+  "moved": [],
+  "failed": [],
+  "warnings": [],
+  "sizeDelta": 40772, // positive or negative, depending on sum
+  "elapsed": 535
+}
+```
+
+#### `npm install --parseable`
+
+The `--parseable` tab-delimited format will remain unchanged by default, to
+prevent potentially breaking parsers. Instead, the `--long`/`-l` option will be
+supported, which will add two additional columns next to each entry: the tarball
+`size`, and the `unpackedSize`. It's up to users to add the numbers up
+themselves.
+
+```
+➜ npm i ssri --dry-run --parseable --long
+
+add     ssri    6.0.0   node_modules/ssri 12116 40772
+```
+
+## Rationale and Alternatives
+
+Adding this sort of information directly to the default summary has one very
+important effect: it makes it very visible and obvious when one installs a
+certain package, what sort of effect that package is having. We already report
+such information by printing out the number of installed dependencies, the
+number of dependencies removed, and even the number of contributors involved.
+
+This information, when printed by default, can then be a useful jumping-off
+point for folks to make more conscious decisions about the size of dependencies
+they're installing.
+
+There are, of course, other alternatives (which can still be considered):
+
+* Have the npm website itself calculate and host this information (initially without project-level awareness, but we could extend it by integrating more with projects or using uploaded metadata from newer npm versions). This would be a similar approach to [packagephoia.now.sh](https://packagephobia.now.sh).
+* Create a separate command to do this analysis. If we do this _instead_ of the current proposal, we'd lose the "default information" part, and rely on discoverability to show this information to users. I think this would make the feature much less useful.
+* The biggest drawback of this proposal, though, in my opinion, is that it does not specifically give _bundle size information_. While I don't think there's any way to come up with exact bundle size information without running the specific bundler the user is using, with the user's specific information, on every single install, and somehow doing a size diff based on that, there may be a compromise to be had: npm might be able to detect whether a bundler is being _used_, and then spit out a pre-calculated bundle estimate (with minification and gzip).
```

Comment:
It might be worth noting that https://bundlephobia.com can provide this bundle size.
/cc @pastelsky